app/src/main/java/com/sdp/movemeet/view/chat/ChatActivity.java

```diff
@@ -123,7 +123,11 @@ protected void onCreate(Bundle savedInstanceState) {
         messageManager = new FirebaseDBMessageManager(database, new MessageSerializer());
 
         fAuth = FirebaseAuth.getInstance();
-        if (fAuth.getCurrentUser() != null) {
+        // The aim is to block any direct access to this page if the user is not logged in
+        if (fAuth.getCurrentUser() == null) {
+            startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
+            finish();
+        } else {
```

Comment:
Again, that's a good way to handle the Authentication! Well done!

Reply:
Haha thank you ! In addition, in one of the classes, we tried to retrieve the id of the user before even testing if the authentication instance existed which resulted in an error.

Reply:
Oh yes, good catch, thank you for correcting the!